### PR TITLE
feat: Recruitment, Employee Life cycle, Holiday list and leave balance report fixes and enhancements

### PR DESCRIPTION
**Pull request by Anurag810**
TODO:
- [x] Fixes
- [ ] Patches(Not required)
- [ ] Test Cases(Not required)

<h3>Job Opening</h3>
<h4>Traceback/Bug Report</h4> 

- The Job Opening was not getting created according to the staffing plan. Throwing "Job Openings for designation Manager already open or hiring completed as per Staffing Plan 2021"
- Department was not getting fetched from the staffing plan.

<h4>Steps to replicate</h4>

- The Job Opening was not getting created according to the staffing Plan
   * Create Staffing Plan For 2 developers.
   * Create job opening For 2 developers

<h3>Expected Behaviour/Fixes</h3>
Now job opening is as per Staffing plan. The behaviour is due to wrong if condition.
<hr>

<h3>Employee Onboarding</h3>

- Removed unused code for checking Unique Email and for Onboarding status.
- Employee Data is not being fetched if Employee is already Created.
- Fetch Designation and Department from Job title 

<h3>Employee separation</h3>

- Unused code for the status

<h3>Employee Leave Balance summary Report</h3>

- Added indentation for better readability.

<h3>Holiday List </h3>

- Setting notification off for Week off.

### CODE REVIEW

File: erpnext/hr/doctype/employee_onboarding/employee_onboarding.js [hunk 1]
```diff
@@ -40,21 +38,6 @@ frappe.ui.form.on('Employee Onboarding', {
...
-				},
-				callback: function(r) {
-					if (r.message) {
-						frm.set_value('boarding_status', r.message);
-					}
-					refresh_field("boarding_status");
-				}
-			});
-		}
-
```
Comment (sahil28297) [inline erpnext/hr/doctype/employee_onboarding/employee_onboarding.js:57]: why remove this?

File: erpnext/hr/doctype/job_applicant/job_applicant.py [hunk 2]
```diff
@@ -26,20 +26,11 @@ def autoname(self):
 		self.name = " - ".join(keys)
 
 	def validate(self):
-		self.applicant_name = self.first_name + " " + self.last_name
-		self.check_email_id_is_unique()
```
Comment (sahil28297) [inline erpnext/hr/doctype/job_applicant/job_applicant.py:30]: why are we removing this check?

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/employee_onboarding/employee_onboarding.js (hunk 1 above)
Comment (Anurag810) [reply erpnext/hr/doctype/employee_onboarding/employee_onboarding.js:57]: Suppose Hr changed the status of Employee-onboarding but on refresh, it was fetching the boarding status as per project and setting the value leaving the form to Unsaved state. which is not good UI.
I checked there is already a server-side validation for completion. making this command useless and causing confusion in the users mind.

File: erpnext/hr/doctype/job_applicant/job_applicant.py (hunk 2 above)
Comment (Anurag810) [reply erpnext/hr/doctype/job_applicant/job_applicant.py:30]: There is a property in DocField to set the field unique. as this is an outdated code so, I removed that and checked the property to is unique. Please refer to the screenshot I had made it unique from here.
Screenshot from 2021-09-22 12-08-02
.

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**